### PR TITLE
[IOTDB-3200] [PartitionInfo] replace bytebuffer with stream

### PR DESCRIPTION
**Pull request by cigarl**
This PR has:
- [x] Replace bytebuffer with stream in `PartitionInfo`.
- [x] Fixed incorrect serialization of multiple sub-maps in `dataPartitionMap`
- [x] Resolve snapshot temporary file deletion failure.

### CODE REVIEW

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/DataPartition.java [hunk 1]
```diff
@@ -248,75 +245,76 @@ public void createDataPartition(
...
+          timePartitionSlotListEntry.getKey().write(protocol);
+          ReadWriteIOUtils.write(timePartitionSlotListEntry.getValue().size(), dataOutputStream);
+          timePartitionSlotListEntry
+              .getValue()
+              .forEach(
+                  x -> {
+                    try {
+                      x.write(protocol);
+                    } catch (TException e) {
+                      throw new RuntimeException(e);
```
Comment (CRZbulabula) [inline node-commons/src/main/java/org/apache/iotdb/commons/partition/DataPartition.java:271]: You can delete this.

**Review by CRZbulabula — COMMENTED** [review]
PTAL.

File: node-commons/src/main/java/org/apache/iotdb/commons/partition/DataPartition.java (hunk 1 above)
Comment (cigarl) [reply node-commons/src/main/java/org/apache/iotdb/commons/partition/DataPartition.java]: fixed

**Review by CRZbulabula — COMMENTED** [review]
Don't forget to clean up redundant snapshot files when takeSnapshot returns false.

**Comment by cigarl** [comment]
> Don't forget to clean up redundant snapshot files when takeSnapshot returns false.

Each module that operates on snapshots, such as `partitionInfo` and `clusterSchema`, already deleted temporary files with the finally clause.  There is no need to clean up again in `ConfigRequestExecutor`.

**Review by CRZbulabula — APPROVED** [review]
LGTM.

**Review by qiaojialin — APPROVED** [review]

**Pull request merged**